include/userobjects/BooleanComboClusteringUserObject.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "ClusteringHeuristicUserObjectBase.h"
```

Comment:
I'm not sure you need to include this here, since you only refer to the class in the header. It may be preferable to forward declare the class and then only include the header in the implementation file.